pkg/policy/distillery_test.go

```diff
@@ -1831,3 +1836,203 @@ func Test_EnsureEntitiesSelectableByCIDR(t *testing.T) {
 		})
 	}
 }
+
+func mapStateAllowsKey(ms *mapState, key Key) bool {
+	var ok bool
+	ms.denies.trie.Ancestors(key.PrefixLength(), key,
+		func(_ uint, _ bitlpm.Key[types.Key], is IDSet) bool {
+			if _, exists := is.ids[identity.NumericIdentity(key.Identity)]; exists {
+				ok = true
+			}
+			return true
+		})
+	if ok {
+		return false
+	}
+	ms.allows.trie.Ancestors(key.PrefixLength(), key,
+		func(_ uint, _ bitlpm.Key[types.Key], is IDSet) bool {
+
+			if _, exists := is.ids[identity.NumericIdentity(key.Identity)]; exists {
+				ok = true
+			}
+			return true
+		})
+	return ok
+}
+
+func TestEgressPortRangePrecedence(t *testing.T) {
+	td := newTestData()
+	identityCache := identity.IdentityMap{
+		identity.NumericIdentity(100): labelsA,
+	}
+	td.sc.UpdateIdentities(identityCache, nil, &sync.WaitGroup{})
+	identity := identity.NewIdentityFromLabelArray(identity.NumericIdentity(100), labelsA)
+
+	type portRange struct {
+		startPort, endPort uint16
+		isAllow            bool
+	}
+	tests := []struct {
+		name       string
+		rules      []portRange
+		rangeTests []portRange
+	}{
+		{
+			name: "deny range (1-1024) covers port allow (80)",
+			rules: []portRange{
+				{80, 0, true},
+				{1, 1024, false},
+			},
+			rangeTests: []portRange{
+				{79, 81, false},
+				{1023, 1025, false},
+			},
+		},
+		{
+			name: "deny port (80) in broader allow range (1-1024)",
+			rules: []portRange{
+				{80, 0, false},
+				{1, 1024, true},
+			},
+			rangeTests: []portRange{
+				{1, 2, true},
+				{79, 0, true},
+				{80, 0, false},
+				{81, 0, true},
+				{1023, 1024, true},
+				{1025, 1026, false},
+			},
+		},
+		{
+			name: "wildcard deny (*) covers broad allow range (1-1024)",
```

Comment:
I do not see the difference with the case below 🧐

Reply:
Good catch.